### PR TITLE
CORE-868: fixed an editing error by completely refactoring buildNavig…

### PR DESCRIPTION
**Pull request by slr71**
…ationRouteRegexp

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -24,22 +24,17 @@ function escapeRegExp(string) {
...
-                routeAlternation = v;
-            } else {
-                routeAlternation = `${routeAlternation}|${NavigationConstants[k]}`;
-            }
-        }
-    }
+    let routeAlternation = Object.entries(NavigationConstants)
+        .filter(([k]) => k !== "LOGIN" && k !== "LOGOUT")
+        .map(([_, v]) => escapeRegExp(v))
+        .join("|");
```
Comment (psarando) [inline src/server/index.js]: 👍 ✨

File: src/server/index.js [hunk 2]
```diff
@@ -24,22 +24,17 @@ function escapeRegExp(string) {
...
 
     // Build and compile the full regular expression.
     return new RegExp(`^/(${routeAlternation})`);
 }
 
+console.log(buildNavigationRouteRegexp());
+
```
Comment (psarando) [inline src/server/index.js]: Debugging code?

Comment (slr71) [reply src/server/index.js]: 🤦‍♀ Yep.

**Comment by slr71** [comment]
Thanks for the reviews!

**Pull request merged**